Added configurations for the addition of letters under buttons, as well as other relevant configurations. Merging does not affect previous builds

alexanderma-okta opened the pull request:
# Summary of changes

The goal of the changes is to add letters to the keypad under the numbers, as well as make some small changes, such as making the numbers bold and the color behavior of the back button (it used to turn the delete button white after clicking the delete button once, but now it will flash a predetermined color and then go back to the default color).

# Changes to Implementation

Each button used to be a "Button" in Android XML but I changed it to a LinearLayout to accommodate both the text and numbers in the button. I also split the variables **textColor** and **textSize** into **numberTextColor**, **lettersTextColor**, **numberTextSize**, **lettersTextSize**, and made appropriate addition of functions/changes to the code to reflect this.

Each button's text used to be set to a number depending on its position. I leveraged this to set each position a specified text as well.

# XML property changes

### Deprecated (but still compatible) XML Configurations :

- keypadTextColor: Sets color of both the numbers and the letters
- keypadTextSize: Sets size of both the numbers and letters

### Added XML Configurations:
- keypadUseDeprecatedColorOptions: Use the deprecated options on color and size over new options (set to false to use new ones)
- keypadNumbersTextColor: Sets the color of the numbers
- keypadNumbersTextSize: Sets the font size of the numbers
- keypadLettersTextColor: Sets the color of the letters
- keypadLettersTextSize: Sets the font size of the letters
- keypadShowLetters: Changes the visibility of the letters
- keypadNumbersBold: Sets the boldness of the numbers text
- keypadLettersBold: Sets the boldness of letter text
- keypadDeleteButtonColor: Sets the color of the delete button
- keypadDefaultDeleteColor: Set to true if you change the color of the delete button
# Snapshots of changes

## Before

![promo](https://user-images.githubusercontent.com/28817207/27564791-fc1328e6-5a8e-11e7-9deb-b4010504154d.png)

## After

<img width="322" alt="screen shot 2017-06-26 at 4 26 02 pm" src="https://user-images.githubusercontent.com/28817207/27564373-08303446-5a8c-11e7-8772-039724b470e4.png"> <img width="321" alt="screen shot 2017-06-26 at 4 49 25 pm" src="https://user-images.githubusercontent.com/28817207/27564826-43f81590-5a8f-11e7-9eac-70abc7659451.png">

alexanderma-okta committed:
Added configurations to add letters to buttons with other relevant configurations such as the colour of the letters. Added configuration to make the font bold.

alexanderma-okta committed:
Merge pull request #1 from okta/add-letters-config

Adding letters to buttons along with other UI tweaking configurations (backwards compatible)

aritraroy commented:
Sorry for getting so late to you. The PR seems good at a glance. But there are few things that I need to decide if I want to include or not. Will get back to it very soon.

aritraroy commented:
Also wanted to understand the motivation behind adding letters under the numbers. Is it just for cosmetic purposes? Or does it serve some real use case?

alexanderma-okta committed:
Minor tweaks to keypad

alexanderma-okta committed:
Fixed delete button- it used to not be responsive when the PinLockView object was set to GONE then VISIBLE, and fixed the centering of the image in the button

alexanderma-okta committed:
wip

alexanderma-okta committed:
Update README.md

alexanderma-okta committed:
Update README.md

alexanderma-okta committed:
Update README.md

alexanderma-okta committed:
Fixing delete button

kavithachandramohan-okta committed:
Layout xml needs to be updated
Layout xml under v21 was updated but we had to update the
layout xml too, since this has old logic of button

helenchen-okta committed:
Update bold typeface

Commit:
Added enter button and attribute to swap enter and delete buttons

Commit:
PinLength fix when setting it programmatically

Commit:
Fixed delete and enter button scaling and alignment

chandrashirashyad-okta commented:
@aritraroy, This came from our UX designer. It is a standard practice as in phone number pad. Some people have hard time remembering PIN numbers. They can remember their PIN by translating the pin to some real word. Words are easier to remember.

By the way, thanks for sharing this wonderful library.

alexanderlin-okta commented:
@aritraroy @tarslangil What version of the library do you plan on including these changes?

tarslangil commented:
@alexanderlin-okta Could you add your version of library to any gradle repository? As i understand this library is not maintained any more.

alexanderlin-okta committed:
Merge pull request #1 from aritraroy/master

Update to latest PinLockView

alexanderlin-okta committed:
Add enter and delete button

alexanderlin-okta committed:
Merge branch 'saltisa-master'

alexanderlin-okta committed:
Add input field, separate delete button, and enter button changes

hansreichenbach-okta committed:
Merge pull request #6 from alexanderlin-okta/dev-temp

Update PinLockView to support enter button, input field, and inline delete button
